src/components/TextInputComponent/TextInputComponent.tsx

```diff
@@ -0,0 +1,72 @@
+import { forwardRef, type InputHTMLAttributes, type KeyboardEventHandler, useMemo } from 'react';
+import './TextInputComponent.css';
+
+interface Props extends InputHTMLAttributes<HTMLInputElement> {
+	label: string;
+	error?: string;
+	helper?: string | null;
+	labelSlot?: React.JSX.Element;
+}
+
+const TextInputComponent = forwardRef<HTMLInputElement, Props>(({
+	id,
+	name,
+	label,
+	disabled,
+	error,
+	helper,
+	value,
+	labelSlot,
+	onKeyDown,
+	...rest
+}, ref) => {
+	// Conditional flag checks
+	if (!label) {
+		console.warn('Custom Warning: input must always have a label. Otherwise it fails the WCAG SC 3.3.2 standard.');
+	}
+
+	const elementId = useMemo(() => id ?? `input-${Math.trunc(Math.random() * 10000).toString(16)}`, [id]);
```

Comment:
The primary purpose of useMemo is "performance optimization".  I understand that's not why we're using it here, but would it not be a better case here, for something like useRef?

Reply:
That was a change I made. We do depend on `id` because it can potentially be set by the user or be changed by the user.

The use of `useMemo` here is not for "expensive computation" but more for caching the `Math.random` result.

We could in theory have the variable just like this:
```ts
const elementId = id ?? `input-${Math.trunc(Math.random() * 10000).toString(16)}`;
```

This means if the consumer of the component did not provide any id, we generate a random string to use. The problem then is that every time the component re-renders the random string will change.

With `useMemo`, the random part is cached and will not be recomputed.